PumaGrasshopper/Utils.cs

```diff
@@ -322,5 +322,21 @@ public static string GetCastErrorMessage(string attribute, string castTarget)
         {
             return "Could not cast attribute " + attribute + " to " + castTarget;
         }
+
+        public static KeyValuePair<string, object> ParseInputPair(IGH_Goo input)
+        {
+            if (!input.CastTo(out string value)) throw new Exception("Could not cast input pair to string");
+
+            string[] tokens = value.Split(':');

```

Comment:
What happens if the string itself contains a ":"? Eg description:"this is the attribute value: with a colon"

Reply:
Are semicolons allowed in cga rule attributes ?

Probably not, right ? If they are not allowed I think it is fine like that.

If they are allowed, then indeed Something could go wrong.